.github/workflows/flutter.yaml

```diff
@@ -20,23 +20,68 @@ jobs:
         with:
           channel: 'stable'
       
+      - name: Check flutter sdk version
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: flutter --version
+      
       - name: Get dependencies
         if: steps.check_files.outputs.files_exists == 'true'
         run: flutter pub get
+      
+      - name: Setup Dart Code Metrics
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: dart pub add --dev dart_code_metrics
 
-      - name: Run static code analysis
+      - name: Dart Code Metrics for lib folder
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: invertase/github-action-dart-analyzer@v1
-        with:
-          fatal-infos: true
+        run: |
+          if [ -d lib ]
+          then
+            dart run dart_code_metrics:metrics analyze lib \
+              --fatal-warnings \
+              --fatal-performance \
+              --fatal-style
+            dart run dart_code_metrics:metrics check-unused-files lib \
+              --fatal-unused
+            dart run dart_code_metrics:metrics check-unused-code lib \
+              --fatal-unused
+          else
+            echo "Lib folder is not found."
+          fi
 
-      - name: Dart Code Metrics
+      - name: Dart Code Metrics for test folder
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: dart-code-checker/dart-code-metrics-action@v2
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          fatal_warnings: true
-          fatal_style: true
+        run: |
+          if [ -d test ]
+          then
+            dart run dart_code_metrics:metrics analyze test \
+              --fatal-warnings \
+              --fatal-performance \
+              --fatal-style
+            dart run dart_code_metrics:metrics check-unused-files test \
+              --fatal-unused
+            dart run dart_code_metrics:metrics check-unused-code test \
+              --fatal-unused
+          else
+            echo "Test folder is not found."
+          fi
```

Comment:
You can build an argument list conditionally instead of copying over the whole command.
Something like

```shell
dirs_to_analyse="lib"
if [ -d test ]; then
  dirs_to_analyse+=" test"
fi

dcm analyse "$dirs_to_analyse".....
```

Reply:
Great suggestion!